packages/flutter/lib/src/cupertino/segmented_control.dart

```diff
@@ -0,0 +1,497 @@
+    if (oldChild != null) {
+      dropChild(oldChild);
+      slots.remove(oldChild);
+    }
+    if (newChild != null) {
+      slots.insert(slot, newChild);
+      adoptChild(newChild);
+    }
+    return newChild;
+  }
+
+  Iterable<RenderBox> get _children sync* {
+    for (int i = 0; i < boxes.length; i += 1) {
+      if (boxes.elementAt(i) != null) {
+        yield boxes.elementAt(i);
+      }
+    }
+  }
+
+  @override
+  void attach(PipelineOwner owner) {
+    super.attach(owner);
+    for (RenderBox child in _children) {
+      child.attach(owner);
+    }
+  }
+
+  @override
+  void detach() {
+    super.detach();
+    for (RenderBox child in _children) {
+      child.detach();
+    }
+  }
+
+  @override
+  void redepthChildren() {
+    _children.forEach(redepthChild);
+  }
+
+  @override
+  void visitChildren(RenderObjectVisitor visitor) {
+    _children.forEach(visitor);
+  }
+
+  @override
+  double computeMinIntrinsicWidth(double height) {
+    double total;
+    for (int i = 0; i < boxes.length; i += 1) {
+      total += _minWidth(boxes.elementAt(i), height);
+    }
+    return total;
+  }
+
+  @override
+  double computeMaxIntrinsicWidth(double height) => computeMinIntrinsicWidth(height);
+
+  @override
+  double computeMinIntrinsicHeight(double width) {
+    return _minHeight(boxes.elementAt(0), width);
+  }
+
+  @override
+  double computeMaxIntrinsicHeight(double width) => computeMinIntrinsicHeight(width);
+
+  @override
+  double computeDistanceToActualBaseline(TextBaseline baseline) {
+    return boxes.elementAt(0).getDistanceToActualBaseline(baseline);
+  }
+
+  @override
+  void performLayout() {
+    double maxHeight = 28.0;
+    Constraints parentConstraints = constraints.loosen();
+    for (int i = 0; i < boxes.length; i += 1) {
+      boxes.elementAt(i).layout(parentConstraints, parentUsesSize: true);
+      double boxHeight = boxes.elementAt(i).size.height;
+      if (boxHeight > maxHeight) {
+        maxHeight = boxHeight;
+      }
+    }
+    final double overallWidth = constraints.maxWidth;
+    final Size overallSize = constraints.constrain(Size(overallWidth, maxHeight));
+
+    Offset centerLayout(Size boxSize, double x) {
+      assert(maxHeight >= boxSize.height);
+      Offset boxOffset;
+      TextDirection textDirection = TextDirection.ltr; //TODO: Change to passing text direction down
+      switch (textDirection) {
+        case TextDirection.rtl:
+          boxOffset = new Offset(x - boxSize.width, (maxHeight - boxSize.height) / 2.0);
+          break;
+        case TextDirection.ltr:
+          boxOffset = new Offset(x, (maxHeight - boxSize.height) / 2.0);
+          break;
+      }
+      return boxOffset;
+    }
+
+    const double left = 0.0;
+
+    final double childWidth = overallSize.width / boxes.length;
+
+    for (int i = 0; i < boxes.length; i += 1) {
+      boxes.elementAt(i).layout(new BoxConstraints.tightFor(width: childWidth, height: maxHeight),
```

Comment:
Create the box constraints outside of the loop, and use the other kind of loop:
```dart
final BoxConstraints childConstraints = new BoxConstraints.tightFor(width: childWidth, height: maxHeight);
for (RenderBox box in boxes) {
  box.layout(childConstraints);
}
```
You might even use `boxes.forEach`.